src/software/world/robot_state.h

```diff
@@ -56,6 +56,48 @@ class RobotState
      */
     AngularVelocity angularVelocity() const;
 
+    /**
+     * Returns if the ball is in the break beam
+     *
+     * @return ball in break beam
+     */
+    bool ballInBeam() const;
```

Comment:
Not super sure how I feel about having this "hardware" information in the `Robot` classes and `AI`. We lost our slack conversation about this and I don't remember exactly what the consensus was. I'd prefer not to have this information in the robot, because I think we thought most could be consumed and hidden in sensor fusion.

We might want to call about this again (and write it down)

Reply:
Maybe I'm misunderstanding but I think it's just the wording of this that makes it "hardware" information. I think naming it "hasBall" would be better.

Reply:
I wonder what additional filtering or processing SensorFusion would do to this information

Reply:
Aaaand this relates to how I don't think a binary sensor should be handled in software. We should chat with elec about getting a latching circuit for the break beam.

Reply:
If this is in the software robot state we should be able to trust that this signal is an accurate indicator that the ball is in front of the robot. I think the name should not have any reference to _what_ hardware was used to make that determination. We could rename it to `hasPossessionOfBall`, but `Possession` is suggestive of a different semantic concept (namely having control of the ball), so I think calling this `ballDirectlyInFrontOfRobot` with an ascii diagram in the jdoc is a better option.

Reply:
Is ballOnDribbler too specific?

Reply:
I've always liked `ballInMouth`. It tells a location without guarantees about controlling the ball

Reply:
> Is ballOnDribbler too specific?

If I saw that name I would assume we have control of the ball. If we do something like `ballDirectlyInFrontOfRobot` like Gareth suggested, there may be a small buffer meaning we _technically_ may not have control of the ball if this function is true. So it depends on exactly what we want this function to mean.

The other part of my concern with putting this in `RobotState` is that I guess I don't see it as something that's part of the robot's literal state (if we abstract away the hardware meaning of breakbeam). Based on how the `RobotState` is currently used and propagated, I think correctly setting up `ballOnDribbler` will be difficult / not maintainable. It will be like why we removed `Timestamp` and split the classes. We were mocking too much unneeded information.

I think I'd prefer a separate module / state machine that tracks robot possession, that can be passed around and queried if we need that information. That way it's separate from state and should be easier to mock for tests. Thoughts? (The difficulty would still be getting this information from SensorFusion to AI, which is where I think this module would live).

Reply:
> I've always liked ballInMouth. It tells a location without guarantees about controlling the ball

I like this. :+1:

Reply:
sounds good